src/components/Form/Form.js

```diff
@@ -58,21 +46,33 @@ export function Form({ children, config, render }) {
       await config.onSubmit(getInputValues())
       alert(
         'form submission successful with values:' +
-          JSON.stringify(getInputValues())
+          JSON.stringify(getInputValues()) +
+          "\n\nBut try again, it's random"
       )
     } catch (e) {
       setFormErrorMessage('Something bad happened: ' + e.toString())
     }
   }
 
+  function input(inputName) {
+    const inputConfig = config.inputs[inputName]
+    return ComponentMap(inputConfig.componentName, {
```

Comment:
Ok, I see how your using this. So...

`ComponentMap` to my mind should be controlled by the client. This way, the client can assemble just the "UI widgets" it needs and circumnavigates my concerns about maintaining a big switch case list of components. So, I have a form and I want a `TextInput` and `ButtonSelectGroup`...in that case, I can assemble my own component map on `config.map` (and even validators with `config.validators`).

I think you could do ^ without really changing the spirit of what you're doing here, no?

If I had any mandate, I'd say inverting the control on the maps and validators is my biggest concern.